File: code/community/1298601623001501697/deepgram-medical-model-mvp.py
import os
from deepgram import Deepgram
import asyncio

API_KEY = os.getenv('DEEPGRAM_API_KEY')
MODEL = 'nova-2-medical'

async def transcribe():
    dg_client = Deepgram(API_KEY)

    audio_url = '<YOUR_AUDIO_URL_HERE>'  # Replace this with your audio file URL
    source = {"url": audio_url}

    response = await dg_client.transcription.prerecorded(  # Change this method according to the real-time requirement
        source,
        {
            "model": MODEL,
            "language": "en-US"
        }
    )

    print("Transcript:", response['results']['channels'][0]['alternatives'][0]['transcript'])

if __name__ == '__main__':
    asyncio.run(transcribe())

File: code/community/1298601623001501697/deepgram-medical-model-mvp.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/deepgram-devs/deepgram-go-sdk/deepgram"
)

func main() {
	dgKey := os.Getenv("DEEPGRAM_API_KEY")
	model := "nova-2-medical"
	ctx := context.Background()

	client := deepgram.NewClient(dgKey)
	audioURL := "<YOUR_AUDIO_URL_HERE>" // Replace with your audio file URL

	options := deepgram.PreRecordedOptions{
		Model:   model,
		Language: deepgram.LanguageEnglishUS,
	}

	result, err := client.TranscribePreRecorded(ctx, audioURL, options)
	if err != nil {
		fmt.Println("Error transcribing audio:", err)
		return
	}

	transcript := result.Results.Channels[0].Alternatives[0].Transcript
	fmt.Println("Transcript:", transcript)
}